processing/src/main/java/io/druid/query/groupby/resource/GroupByQueryBrokerResource.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.groupby.resource;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.metamx.common.logger.Logger;
+import io.druid.collections.ResourceHolder;
+
+import java.io.Closeable;
+import java.nio.ByteBuffer;
+import java.util.List;
+
+/**
+ * This class contains all resources required by the Broker during executing a group-by query.
+ * Currently, it contains only merge buffers, but any additional resources can be added in the future.
+ */
+public class GroupByQueryBrokerResource implements Closeable
+{
+  private static final Logger log = new Logger(GroupByQueryBrokerResource.class);
+
+  private final ResourceHolder<List<ByteBuffer>> mergeBuffersHolder;
+  private final List<ByteBuffer> mergeBuffers;
+
+  public GroupByQueryBrokerResource()
```

Comment:
This default constructor is used when any merge buffers are not required for groupBy execution.

I renamed to ```GroupByQueryResource``` which is more general name because druid's convention generally doesn't distinguish the broker side things and others. 
```GroupByQueryResource``` can be used by queryable nodes, i.e., brokers, historicals, and real times. However, currently it is used by only brokers to get merge buffers atomically if necessary. And, even in brokers, merge buffers are not used when the groupBy strategy v1 is used.